src/components/CodeSelectionMenu.tsx

```diff
@@ -114,25 +124,36 @@ const CodeSelectionMenu: React.FC<CodeSelectionMenuProps> = ({
     );
   };
 
-  const handleClose = () => {
-    if (abortController) {
-      abortController.abort();
-      setAbortController(null);
-    }
-    setIsExplaining(false);
-    setShowExplanation(false);
-    onClose();
+  const popupPosition = {
+    left: Math.max(10, Math.min(position.x, window.innerWidth - 350)),
```

Comment:
```suggestion
    left: Math.max(300, Math.min(position.x, window.innerWidth - 350)),
```
To make sure that popup has some considerable left spacing and doesn't entirely show on top of the selected code.